tests/by-util/test_cksum.rs

```diff
@@ -2668,6 +2668,97 @@ mod gnu_cksum_c {
     }
 }
 
+#[test]
+fn test_sha3_with_length() {
+    // Test SHA3-224
+    new_ucmd!()
+        .arg("-a")
+        .arg("sha3")
+        .arg("--length")
+        .arg("224")
+        .arg("lorem_ipsum.txt")
+        .succeeds()
+        .stdout_contains("SHA3-224 (lorem_ipsum.txt) = ");
+
+    // Test SHA3-256
+    new_ucmd!()
+        .arg("-a")
+        .arg("sha3")
+        .arg("--length")
+        .arg("256")
+        .arg("lorem_ipsum.txt")
+        .succeeds()
+        .stdout_contains("SHA3-256 (lorem_ipsum.txt) = ");
+
+    // Test SHA3-384
+    new_ucmd!()
+        .arg("-a")
+        .arg("sha3")
+        .arg("--length")
+        .arg("384")
+        .arg("lorem_ipsum.txt")
+        .succeeds()
+        .stdout_contains("SHA3-384 (lorem_ipsum.txt) = ");
+
+    // Test SHA3-512
+    new_ucmd!()
+        .arg("-a")
+        .arg("sha3")
+        .arg("--length")
+        .arg("512")
+        .arg("lorem_ipsum.txt")
+        .succeeds()
+        .stdout_contains("SHA3-512 (lorem_ipsum.txt) = ");
+}
+
+#[test]
+fn test_sha3_invalid_length() {
+    // Test SHA3 with invalid length (not 224, 256, 384, or 512)
+    new_ucmd!()
+        .arg("-a")
+        .arg("sha3")
+        .arg("--length")
+        .arg("128")
+        .arg("lorem_ipsum.txt")
+        .fails()
+        .stderr_contains("digest length for 'SHA3' must be 224, 256, 384, or 512");
+
+    // Test SHA3 without length
+    new_ucmd!()
+        .arg("-a")
+        .arg("sha3")
+        .arg("lorem_ipsum.txt")
+        .fails()
+        .stderr_contains("--algorithm=sha3 requires specifying --length 224, 256, 384, or 512");
+}
```

Comment:
This functionality is already tested by `test_sha_length_invalid` and `test_sha_missing_length`.
```suggestion
```